### ISSUE TITLE
Suggestion for Neural Machine Translation with Attention tutorial

### ISSUE THREAD

**Issue by jasonzhang2022** [tensorflow/tensorflow#28371]
<em>Please make sure that this is a documentation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:doc_template</em>

**System information**
- TensorFlow version:
2.0
- Doc Link:
https://www.tensorflow.org/alpha/tutorials/text/nmt_with_attention

**Describe the documentation issue**

1. max_length function is not needed. The keras.preprocessing.sequence.pad_sequences already pad the sequence to max length. One can just get max length by input_seqs.shape[1]
2. need to explain why we skip 0 for "convert" and "loss_function". I think it is because of padding
3. Validation set is not used.  This could be an good example to explain how we evaluate the model.
4.  GRU unit in encoder,  Dense in Attention weight calculation,  GRU unit in decoder all have the same units parameter. This is not necessary.  A note could be added to explain that same unit is used for convenience.

**We welcome contributions by users. Will you be able to update submit a PR (use the [doc style guide](https://www.tensorflow.org/community/documentation)) to fix the doc Issue?**

Yes

**Comment by qlzh727** [comment]
Assigning to Mark who in charge of doc and colab. I think all the suggestions here make sense and would improve the readability for the colab.

**Comment by yashk2810** [comment]
These updates sound great. Can you please send a PR here (https://github.com/tensorflow/docs/blob/master/site/en/r2/tutorials/text/nmt_with_attention.ipynb)?

**Comment by jvishnuvardhan** [comment]
@jasonzhang2022 I think [this tutorial](https://www.tensorflow.org/alpha/tutorials/text/nmt_with_attention) was updated based on your suggestions. I am closing this issue as it was resolved. Please feel free to reopen if I am mistaken. Thanks!
